src/Http/WebApp/WebAppView.php

```diff
@@ -293,13 +293,37 @@ public function render(Request $request)
         $baseUrl = array_get($forum, 'data.attributes.baseUrl');
         $view->cssUrls = $this->buildCssUrls($baseUrl);
         $view->jsUrls = $this->buildJsUrls($baseUrl);
+        $this->addLinkHeaders();
 
         $view->head = implode("\n", $this->head);
         $view->foot = implode("\n", $this->foot);
 
         return $view->render();
     }
 
+    protected function addLinkHeaders()
+    {
+        $css = [
+            str_replace(public_path(), '', $this->getCss()->getFile()),
+            str_replace(public_path(), '', $this->localeCss->getFile())
+        ];
+        $js = [
+            str_replace(public_path(), '', $this->getJs()->getFile()),
+            str_replace(public_path(), '', $this->localeJs->getFile())
+        ];
+        foreach ($css as $cssUrl) {
+            if ($cssUrl != null) {
+                header('Link: <'.$cssUrl.'>; rel=preload, as=style', false);
+            }
+        }
+
+        foreach ($js as $jsUrl) {
+            if ($jsUrl != null) {
+                header('Link: <'.$jsUrl.'>; rel=preload, as=script', false);
```

Comment:
I was originally going to make a middleware for this but I couldn't seem to figure out how that works. Would actually be better as a middleware I think because we could scan the entire page for images and all assets to push. Instead of just the ones defined here.